core/src/main/c/share/vect.cpp

```diff
@@ -65,10 +71,32 @@
 
 #define SUM_INT F_SSE2(sumInt)
 
+#define HAS_NULL F_SSE2(hasNull)
+
 #else
 
 #endif
 
+#ifdef HAS_NULL
+
+bool HAS_NULL(int *pi, long count) {
+    const int step = 4;
+    const int remainder = (int) (count - (count / step) * step);
+    const int *vec_lim = pi + count - remainder;
+
+    Vec4i vec;
+    for (; pi < vec_lim; pi += step) {
+        vec.load(pi);
+        if (horizontal_find_first(vec == -1)) {
+           return true;
+        }
+    }
+    return false;
```

Comment:
there need to be code that deals with remaining data, in case count is not divisible by 4 exactly